### ISSUE TITLE
Upgrade @pubsweet/ui and hopefully fix deprecated use of attr({}) 

### ISSUE THREAD

**Issue by Cooryd** [elifesciences/elife-xpub#2235]
## Background

The current version of `@pubsweet/ui` we are using has a `Menu` atom being used on the details step of the submission wizard. The use of this is throwing this warning: 

```
Functions as object-form attrs({}) keys are now deprecated and will be removed in a future version of styled-components. Switch to the new attrs(props => ({})) syntax instead for easier and more powerful composition. The attrs key in question is "aria-selected" on component "styled.div".
```

use of this becomes obsolete in the most recent version of `styled-components` so we will need to upgrade `@pubsweet/ui` before we can update `styled-components`

### FYI

Upgrading too `@pubsweet/ui@^10.3.6` doesn't seem to be straight forward as `babel` errors are thrown when trying to compile after upgrading.

**Comment by petereast** [comment]
@diversemix  and I have been working on this today, we've made a little bit of progress by
- updating everything `babel` related to `v7`
- moving config into `webpack/rules.js` so it's actually used in the right context
- swearing a lot and crying a bit

After doing this, we've managed to get `webpack` to build successfully, but when you load the app in the browser, it just straight up doesn't work, throwing a `TypeError: cannot write to exports`. The next step in looking into this is to have a look at this https://stackoverflow.com/questions/35983376/babel-loader-not-recognizing-es2015-code

**Comment by petereast** [comment]
I'm throwing in the towel for this one as this is not a priority right now- in #2239 I managed to get the application to render again, but then `jest` completely failed. I'm calling on the help of @jure to see how pubsweet has internally solved this issue.

**Comment by jure** [comment]
Happy to take a look at this and #2239! The end result should be good, faster builds, less webpack config, etc.

**Comment by petereast** [comment]
That sounds great! Feel free to give me a shout if you need anything - I'm quite curious about how it'll actually work
